### DISCUSSION TITLE
Custom models don't retrieve the content of File columns

### DISCUSSION THREAD

**Discussion by ilearnio** [pocketbase/pocketbase#2712] [Q&A]
I just created a custom model with an "audioFile" column, for which I tried to set a type of `string` or `any` but in both cases it just retrieves an empty value from DB when fetched, even though I'm pretty sure that there's a file stored in DB, I can see if from Pocketbase admin. The rest of the columns of different types are displayed properly.

Here's a shortened example of how my model looks like:

```go
type Audio struct {
	models.BaseModel
	
	Id        string `json:"id"`
	AudioFile any    `json:"audioFile"`
}

func (m *Audio) TableName() string {
	return "audios"
}
```

And here's how I'm doung a query:

```go
dao := app.Dao()

record := &models.Audio{}

err := dao.ModelQuery(&models.Audio{}).
	AndWhere(dbx.HashExp{"videoId": videoId, "videoQuoteId": videoQuoteId}).
	Limit(1).
	One(record)
if err != nil {
	return nil, err
}
spew.Dump("record", record)
```

And here's the output:

```
(string) (len=6) "record"
(*models.Audio)(0x1400002e2d0)({
 BaseModel: (models.BaseModel) {
  isNotNew: (bool) true,
  Id: (string) "",
  Created: (types.DateTime) 2023-06-14 03:42:57.763Z,
  Updated: (types.DateTime) 2023-06-14 19:35:58.310Z
 },
 Id: (string) (len=15) "m3ejdp5fkcf59dx",
 AudioFile: (interface {}) <nil>
})
```

Am I doing something wrong here?

PS: Using latest PB

**Comment by ganigeorgiev** [comment]
Files are stored on the disk (or in S3 if configured). In the database we store only the file(s) name as record value. So in your case if `audioFile` is a `file` field, then it must be typed as `string` (or `types.JsonArray` if multiple). You can also use the `db` tag to explicitly set the column name, eg:

```go
AudioFile `json:"audioFile" db:"audioFile"`
```

I don't have context to understand the end goal and the use case but note that custom struct models like the above will not work with the `OnRecord*` hook and it may be easier to use the [Record model  and its helpers](https://pocketbase.io/docs/record-methods/) instead. Eventually autogenerated typed models and hooks will be implemented in the future but for now is a low priority (_you could also check the recent discussion in https://github.com/pocketbase/pocketbase/discussions/2697_).

**Reply by ilearnio** [reply]
That worked! Thank you so much @ganigeorgiev!
